feat: Implement channel blocking feature (#12647)

mstudio005 opened the pull request:
### What is it?

- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

### Description of the changes in your PR

- Add `BlockedChannelsManager` utility class for local storage using SharedPreferences  
- Implement automatic filtering of blocked channels in Trending, Search, and Recommendations  
- Add **Block Channel** option in video long-press context menu with undo support  
- Create **Blocked Channels** management screen in Settings → Content  
- Add necessary string resources for UI  
- Update `gradle.properties` to use Java 17 for build compatibility  

### Before/After Screenshots or Screen Record

**Before:**  
![Screenshot_20251112_081656_NewPipe](https://github.com/user-attachments/assets/993d72be-6fad-454e-9020-139d48c27257)
![Screenshot_20251112_081731_NewPipe](https://github.com/user-attachments/assets/fdd83e53-208f-4ddb-b062-ebc550820060)
  

**After:**  
Shows new *“Block Channel”* option in the video overflow menu, and *“Blocked Channels”* section under Settings → Content.
![Screenshot_20251111_235524_NewPipe Debug](https://github.com/user-attachments/assets/ab40b45d-72ef-420f-ae23-2ae69ca0570e)
![Screenshot_20251111_235536_NewPipe Debug](https://github.com/user-attachments/assets/b7a458b7-bc2f-4bab-aed2-b9d1f706bd44)
![Screenshot_20251112_055750_NewPipe Debug](https://github.com/user-attachments/assets/a3323d37-deb7-4eb5-ab97-ed3738dc3529)


### Fixes the following issue(s)
- Fixes #12647

### Relies on the following changes
- None

### APK testing
The APK can be found by going to the **Checks** tab below the title.  
On the left pane, click **CI**, scroll down to **artifacts**, and click **app** to download the debug APK of this PR. This is also tested in Samsung Tablet

### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md)

mstudio005 committed:
feat: Implement channel blocking feature (#12647)

- Add BlockedChannelsManager utility class for local storage using SharedPreferences
- Implement automatic filtering of blocked channels in Trending, Search, and Recommendations
- Add 'Block Channel' option in video long-press context menu with undo support
- Create Blocked Channels management screen in Settings > Content
- Add comprehensive string resources for the feature
- Update gradle.properties to use Java 17 for build compatibility

Features:
* BlockedChannelsManager: Manages blocked channel IDs via SharedPreferences
* InfoListAdapter: Filters out videos from blocked channels automatically
* StreamDialogDefaultEntry: New BLOCK_CHANNEL entry with Snackbar undo
* BlockedChannelsFragment: UI for viewing and managing blocked channels
* Settings integration: Accessible via Settings > Content > Blocked Channels

Tested:
- Debug APK build successful
- All code passes checkstyle and ktlint validation
- Feature follows MVVM architecture and NewPipe coding standards

TobiGr commented on app/src/main/java/org/schabi/newpipe/util/BlockedChannelsManager.kt:
Blocked channels should be stored in the DB with url and name so that the name can be displayed to the user.

TobiGr commented on app/src/main/java/org/schabi/newpipe/info_list/dialog/StreamDialogDefaultEntry.java:
```diff
@@ -135,7 +138,34 @@ public enum StreamDialogDefaultEntry {
+                // Unblock the channel
+                BlockedChannelsManager.INSTANCE.unblockChannel(
+                        fragment.requireContext(), uploaderUrl);
+            });
```
.setActionTextColor(Color.YELLOW)

TobiGr reviewed (changes requested):
Hello and thank you for the PR.

1. Please do not open AI generated PRs. They lack the required info and code quality.
2. Feature requests should target the refactor branch because the app is ondergoing a rewrite and the dev branch is only used for fixes.
3. Please test your changes on different screen sizes (phones, tablets, TV)
4. Ensure that our theme is used, the texts are all readable and the contrast is good.

mstudio005 commented:
**Closing this in favour of a new PR ~#13491~ #13492 targeting the refactor branch as requested by @TobiGr. The feature has been rewritten from scratch addressing all the review feedback.**
